mono/mini/method-to-ir.c

```diff
@@ -11196,80 +11196,6 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 			MONO_ADD_INS (cfg->cbb, ins);
 			break;
 		}
-		case MONO_CEE_MONO_JIT_ATTACH: {
-			g_assert (method->wrapper_type != MONO_WRAPPER_NONE);
```

Comment:
This was optimized code which avoided calling into the runtime if not needed.
Also, it handled the case when calling into the runtime would go though the trampoline code which expected the runtime to be already attached.

Reply:
This is not compatible with cooperative and hybrid suspend which are becoming defaults. And since we want to generate coop-aware code even when the runtime is not configured for cooperative/hybrid (in the AOT case for example), we need this change. Also, the existing `mono_threads_attach_coop` are compatible with the preemptive case as well.